app/services/bulk_upload/sales/year2022/row_parser.rb

```diff
@@ -1110,8 +1110,10 @@ def validate_incomplete_soft_validations
       next if question.completed?(log)
 
       question.page.interruption_screen_question_ids.each do |interruption_screen_question_id|
+        next if log.form.questions.none? { |q| q.id == interruption_screen_question_id && q.page.routed_to?(log, nil) }
```

Comment:
I don't know much about this row parsing, however this sort of function appears to come up a lot, could this be put in a helper method and referred to by multiple questions? even from multiple years perhaps.

Reply:
There is definitely a wider conversation that can be had about whether all these row parser objects should inherit from a general row parser class or include a helper module.
When I suggested that Kat pointed me towards this (which is an interesting short read):
https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction
which is what Phil cited when he made the design choice to have all the row parsers just be their own self-contained complete things.
I'm not sure I agree with that in this case since I think this particular method and a couple of others really are doing the same thing in each row parser and I wouldn't imagine they would start to diverge. However I didn't feel like taking on the tech debt at this stage and Phil is a much more experienced developer than me so he may have had some instinct as to why they may diverge in the future.